### ISSUE TITLE
Timex.Interval.overlaps? not commutative?

### ISSUE THREAD

**Issue by bundacia** [bitwalker/timex#468]
### Steps to reproduce

```elixir
iex(51)> a = Interval.new(from: ~N[2018-01-01 12:00:00], until: ~N[2018-01-10 12:00:00], left_open: true, right_open: true, step: [minutes: 1])
%Timex.Interval{from: ~N[2018-01-01 12:00:00], left_open: true,
 right_open: true, step: [minutes: 1], until: ~N[2018-01-10 12:00:00]}
iex(52)> b = Interval.new(from: ~N[2018-01-01 12:00:00], until: ~N[2018-01-01 13:00:00], left_open: true, right_open: true, step: [minutes: 1])
%Timex.Interval{from: ~N[2018-01-01 12:00:00], left_open: true,
 right_open: true, step: [minutes: 1], until: ~N[2018-01-01 13:00:00]}
iex(53)> Interval.overlaps?(a, b)
false
iex(54)> Interval.overlaps?(b, a)
true
```

### Description of issue

I would expect that `&Timex.Interval.overlaps?/2` would be commutative, that is, the order of the arguments shouldn't matter. If `a` overlaps `b` than is seems like `b` must overlap `a` as well, but as shown in the `iex` session above, this is not always the case. 

Is this a bug, or am I misunderstanding how "overlapping" is defined in this context?

### Versions
Timex version `3.3.0`

Erlang/OTP 20 [erts-9.0.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.5.2

**Comment by ckhrysze** [comment]
Thank you for reporting this! It was in fact an issue, but has been fixed. That fix is in the release of 3.4.1 that went out yesterday.

**Comment by bundacia** [comment]
Thanks!!
